### PR TITLE
refactor(git): Delegate commit function to platform

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- use `platform.commitFiles` that calls `git.commitFiles`
- fix mocks

## Context:

- Ref: #11537
- Blocked by #13821

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
Could we simplify this by having the platform index logic add/insert this function to platforms which don't already have it?

**Comment by zharinov** [comment]
If we make it optional, we'll have to check the function existence before call. If we don't, `gitea` still will have this dummy implementaiton. So maybe we'll need to refactor it first... IMO, these dummy implementations are redundant, but straightforward.

File: lib/platform/azure/index.ts [hunk 1]
```diff
@@ -840,3 +841,10 @@ export async function deleteLabel(
 export function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
   return Promise.resolve([]);
 }
+
+// istanbul ignore next
+export function commitFiles(
```
Comment (viceice) [inline lib/platform/azure/index.ts:846]: Do we really need those simple wrapper?
Can't we do something like `platform.commitFiles ? platform.commitFiles(config) : git.commitFiles(config)` where required?

File: lib/workers/branch/commit.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export function commitFilesToBranch(
     throw new Error(CONFIG_SECRETS_EXPOSED);
   }
   // API will know whether to create new branch or not
-  return commitFiles({
+  return platform.commitFiles({
```
Comment (viceice) [inline lib/workers/branch/commit.ts:50]: We can add a `commit-files.ts` for this function, so we can easier mock in tests.
Related to comment above.

**Comment by zharinov** [comment]
Fixes

File: lib/workers/branch/commit.ts [hunk 3]
```diff
@@ -46,11 +48,16 @@ export function commitFilesToBranch(
...
+  const commitConfig: CommitFilesConfig = {
     branchName: config.branchName,
     files: updatedFiles,
     message: config.commitMessage,
     force: !!config.forceCommit,
-  });
+  };
+
+  // API will know whether to create new branch or not
+  return config.platformCommit && platform.commitFiles
```
Comment (viceice) [inline lib/workers/branch/commit.ts:60]: Should we warn for unsupported platforms?

I would also extract this code for deduplication.

File: lib/workers/branch/commit.ts (hunk 2 above)
Comment (zharinov) [reply lib/workers/branch/commit.ts]: I don't think we need this for now, all required mocks are easy enough

**Review by viceice — COMMENTED** [review]
Otherwise LGTM.

**Comment by rarkins** [comment]
@zharinov please resolve conflicts and mark as ready for review when ready

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
> If we make it optional, we'll have to check the function existence before call. If we don't, `gitea` still will have this dummy implementaiton. So maybe we'll need to refactor it first, but if we refactor one platform module, why not refactor them all to classes... IMO, these dummy implementations are redundant, but straightforward.

Yes, we should class-ify platform code, so we can have abstract base class to share core functions, which as equal to all platforms.

**Pull request merged**